app/models/user.rb

```diff
@@ -15,20 +15,34 @@ class User < ApplicationRecord
 
   attr_accessor :remember_token, :reset_token
 
-  has_many :awards
-  has_many :submissions
-  has_many :participations
+  has_many :submissions, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :notifications, dependent: :destroy
+  # user.followers will return all users that follow THIS user
+  has_many :followers, class_name: 'Follower', foreign_key: 'following_id', dependent: :destroy
+  # user.following will return all users that THIS user follows
+  has_many :following, class_name: 'Follower', dependent: :destroy
+  has_many :user_session, dependent: :nullify
+
+  has_many :challenge_entries, dependent: :destroy
+  has_many :participations, dependent: :destroy
+  has_many :awards, dependent: :destroy
   has_many :badges, through: :awards
   has_many :challenges, through: :participations
-  has_many :comments
-  has_many :moderator_logs
-  has_many :moderator_logs, as: :target
-  has_one :moderator_application
+  has_many :created_challenges,
+           class_name: 'Challenge', foreign_key: 'creator_id', dependent: :nullify
+  has_many :user_feedbacks, dependent: :nullify
+  has_many :discussions, dependent: :nullify
+
+  has_one :moderator_application, dependent: :destroy
+  has_many :site_ban, dependent: :destroy
+
+  has_many :created_logs, class_name: 'ModeratorLog', dependent: :nullify
+  has_many :moderator_logs, as: :target, dependent: :nullify
   has_many :house_participations, dependent: :nullify
-  has_many :followers, class_name: 'Follower', foreign_key: 'following_id', dependent: :destroy
-  has_many :following, class_name: 'Follower', dependent: :destroy
 
   before_save { self.email = email.downcase }
+  before_destroy { throw(:abort) if marked_for_death }
```

Comment:
What's the rationale behind this little bit here?

Reply:
I figured if they're marked for death for some serious reason, they should not be able to delete their account so that punishment can't be dealt and the evidence can be thrown away.